### ISSUE TITLE
custom options in usemin and html file does not work after upgraded to 0.3.11

### ISSUE THREAD

**Issue by hantsy** [zont/gulp-usemin#92]
In my gulpfile, I slipt the js opt to two options in usemin options.

<pre>
usemin{
jsmin:XXX//contact the vendor compressed js lib
js:XXX // do more for project scoped js files
}
</pre>

and in the html file, I used two macros to complete the tasks.

<pre>
&lt;!--build:jsmin js/vendor.js-->
&lt;script ....>
&lt; endbuild-->

&lt;!--build:js js/app.js-->
&lt;script ....>
&lt; endbuild-->

</pre>

After I upgrade to 0.3.11, the jsmin task did not work, but when I downgraded to 0.3.8, everything works well.

**Comment by zont** [comment]
Test for many blocks successful (https://github.com/zont/gulp-usemin/blob/master/test/fixtures/many-blocks.html).
Can you give more data about a bug?

**Comment by cjus** [comment]
@hantsy @zont 

I've encountered what is probably the same issue. I was using 0.3.11 and seeing an issue where my app.js lib wasn't being built.  @hantsy for my issue I needed to roll back to 0.3.6 instead of 0.3.8.

I noticed that the file that was being processed appeared to go through some sort of recursion.  This resulted in the following.  Notice the repeating hash sequences and no app.js

```
┏━━━━━━━➤ [cjus@phoenix]: ~/dev/modern-web-seed/dist/js
┗➤ cd ../js; ll
total 1584
drwxr-xr-x  4 cjus  staff   136B Jan 25 15:19 .
drwxr-xr-x  7 cjus  staff   238B Jan 25 15:19 ..
-rw-r--r--  1 cjus  staff   392K Jan 25 15:19 lib-871301f9-871301f9-bddeb64a-bddeb64a-11a8d805-11a8d805-11a8d805-11a8d805.js
-rw-r--r--  1 cjus  staff   392K Jan 25 15:19 lib-871301f9.js
```

After reverting to 0.3.6 I get:

```
┏━━━━━━━➤ [cjus@phoenix]: ~/dev/modern-web-seed/dist/js
┗➤ cd ../js; ll
total 800
drwxr-xr-x  4 cjus  staff   136B Jan 25 15:29 .
drwxr-xr-x  7 cjus  staff   238B Jan 25 15:29 ..
-rw-r--r--  1 cjus  staff   967B Jan 25 15:29 app-e17cc92c.js
-rw-r--r--  1 cjus  staff   392K Jan 25 15:29 lib-871301f9.js
```
@zont you can recreate this issue by changing the version of gulp-usemin used in the following repo:

https://github.com/cjus/modern-web-seed

to 0.3.11 and running: `gulp dist`

**Comment by zont** [comment]
Fixed in 0.3.14
